src/templates/defaultTemplate.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import type {
+  ChangelogType,
+  VersionChangelogType,
+  TemplateHelpersType,
+  EntryType,
+  EntryKindType
+} from '../types';
+
+const entryKindToReadable = {
+  change: 'Changed',
+  addition: 'Added',
+  fix: 'Fixed'
+};
+
+export default function defaultTemplate(
+  helpers: TemplateHelpersType,
+  changelog: ChangelogType
+) {
+  return [
+    '# Changelog',
+    ...changelog.map((versionChangelog) => renderVersionChangelog(helpers, versionChangelog))
+  ].join('\n\n');
+}
+
+function renderVersionChangelog(
+  helpers: TemplateHelpersType,
+  { version, entries }: VersionChangelogType
+): string {
+  // $FlowFixMe: Object.keys() does not refine correctly based on type definition of keys
+  const entryKeys: EntryKindType[] = Object.keys(entries);
+
+  return [
+    `## Version \`${helpers.stringifyVersion(version)}\``,
+    ...entryKeys.map((entryKind) => renderEntriesOfKind(helpers, entryKind, entries[entryKind]))
+  ].join('\n\n');
+}
+
+function renderEntriesOfKind(
+  helpers: TemplateHelpersType,
+  kind: EntryKindType,
+  entries: EntryType[]
+): string {
+  if (entries.length === 0) {
+    return '';
+  }
+
+  return [
+    `### ${entryKindToReadable[kind]}`,
+    ...entries.map(({ component, description }) => {
```

Comment:
So many reasons:
- I had to cope with the fact that component might be null for which I would have had to add or implement external helpers that cover equality checks
- I wanted to remove the duplication of the different entry kind renderings (I had the same code in place for additions, changes and fixes) -> I had to iterate the `components`-map for that which is incredibly hard to do in Handlebars
- there is no validation (e.g. flow) that I'm rendering something useful
- readability was horrible because I cannot indent the handlebars blocks since that would change the Markdown semantics
- changing the `.hbs` file does not trigger rebuilds automatically since there is no formal dependency to the file
- we do not expose a templating interface right now – once we do that I'd reconsider supporting (another?) templating engine